apps/strapi-cms/src/api/summary-feedback/content-types/summary-feedback/lifecycles.ts

```diff
@@ -0,0 +1,64 @@
+import { upsertTypesenseCollection } from '@george-ai/typesense-client'
+
+const transformAndUpsertFeedback = async (id) => {
+  const summaryFeedbackResult = await strapi.entityService.findOne(
+    'api::summary-feedback.summary-feedback',
+    id,
+    {
+      populate: ['web_page_summary'],
+    },
+  )
+
+  const webPageSummaryResult = await strapi.entityService.findOne(
+    'api::web-page-summary.web-page-summary',
+    summaryFeedbackResult.web_page_summary.id,
+    {
+      populate: ['scraped_web_page', 'summary_feedbacks'],
+    },
+  )
+
+  const updatedAt = new Date(webPageSummaryResult.updatedAt)
+  const summaryFeedbacks = (
+    webPageSummaryResult.summary_feedbacks ?? []
+  ).filter((feedback) => {
+    const createdAt = new Date(feedback.createdAt)
+    return createdAt > updatedAt
+  })
+
+  const popularity = summaryFeedbacks.reduce((accumulator, feedback) => {
+    const vote = feedback.voting
+    if (vote === 'up') {
+      return accumulator + 1
+    }
+    if (vote === 'down') {
```

Comment:
```suggestion
    else if (vote === 'down') { ... }
    else { console.warn('value ${vote} is not implemented for calculating the popularity, feedback.id)
```

Else it disappears if something goes wrong here.